net/httpclient.go

```diff
@@ -368,6 +376,11 @@ func (t *Transport) RoundTrip(req *http.Request) (*http.Response, error) {
 	if t.bearerToken != "" {
 		req.Header.Set("Authorization", "Bearer "+t.bearerToken)
 	}
+	if t.requestTimeout != 0 {
+		reqWithTimeoutCtx, cancelReqContext := context.WithTimeout(req.Context(), t.requestTimeout)
+		req = req.WithContext(reqWithTimeoutCtx)
+		defer cancelReqContext()
```

Comment:
This cancels context before response has been consumed.

https://pkg.go.dev/net/http#Request.WithContext
> For outgoing client request, the context controls the entire lifetime of a request and its response: obtaining a connection, sending the request, and reading the response headers and body.

Reply:
@AlexanderYastrebov thank you for pointing that out, but I'm not sure if we have any ways to cover only request timeout then, I see another method https://pkg.go.dev/net/http#NewRequestWithContext to create a request with context and notice that the behavior of the context is the same there as well, so I'm not sure how to proceed. Do you have any suggestions?

Reply:
I do not see an easy way to use request context inside of the client. But client user may always create request with context and cancel it after consuming the response.

Reply:
@AlexanderYastrebov Yeah seems like. @szuecs do you have any suggestions?

Reply:
I guess you would need to have a `net.NewRequest` that you would then need to use to apply it with a default.
```
type struct O {
defaultDuration time.Duration
}

// func NewRequestWithContext(ctx context.Context, method, url string, body io.Reader) (*Request, error)
func (o *O) NewRequest(ctx context.Context, method, url string, body io.Reader)  (*http.Request, error) {
   return http.NewRequestWithContext( context.WithTimeout(ctx, o.defaultDuration), method, body)
}
```

Maybe `O` is the Client ? But this would be also a bit weird to do `client.NewRequest()`.

Reply:
Do we really need to complicate client when user can easily create and close request with context on their own?

Reply:
@AlexanderYastrebov No, I think you are right. The use needs to do anyways the call and then it's likely better to have a const/var in their code base that they reference to get their "default" request timeout.